ast/src/parsed/mod.rs

```diff
@@ -17,7 +17,12 @@ pub enum PilStatement<T> {
     /// Name of namespace and polynomial degree (constant)
     Namespace(usize, String, Expression<T>),
     PolynomialDefinition(usize, String, Expression<T>),
-    PublicDeclaration(usize, String, PolynomialReference<T>, Expression<T>),
```

Comment:
Public declarations refer to namespaced polynomials like `A[3]` or `B`. Shifts do not make sense in this context, this is now enforced by the type system.